File: 2020/src/solutions/Day2.java
package solutions;

import java.util.*;
import utils.*;

public class Day2 implements Day{
	
	Input input;
	
	public Day2() {
		this.input = new Input("/home/joel/eclipse-workspace/AdventOfCode/input/day2.txt");
	}

	@Override
	public String part1() {
		
		// List<String> strList = input.getList();
		String[] strArr = input.getArray();
		
		int sum = 0;
		for (String s : strArr) {
			String[] splitted = s.split(":");
			String[] info = splitted[0].split("-");
			int lower = Integer.parseInt(info[0]);
			String[] other = info[1].split(" ");
			int higher = Integer.parseInt(other[0]);
			char letter = other[1].toCharArray()[0];
			int count = 0;
			for (char c : splitted[1].toCharArray()) {
				if (c == letter) {
					count++;
				}
			}
			if (count >= lower && count <= higher) {
				sum++;
			}
		}
		
		return String.valueOf(sum);
	}
		
	public String part2() {

		// List<String> strList = input.getList();
		String[] strArr = input.getArray();
		
		int sum = 0;
		for (String s : strArr) {
			String[] splitted = s.split(":");
			String[] info = splitted[0].split("-");
			int i1 = Integer.parseInt(info[0]);
			String[] other = info[1].split(" ");
			int i2 = Integer.parseInt(other[0]);
			char letter = other[1].toCharArray()[0];
			String word = splitted[1].substring(1);
			
			if (word.charAt(i1-1) == letter ^ word.charAt(i2-1) == letter) {
				sum++;
			}
		}
		
		return String.valueOf(sum);
	}
		
		
	}

File: 2022/src/day2.py
import sys


def main():

    input = []
    with open('input/day2.txt') as f:
        for line in f.readlines():
            l = line.replace("\n","")

            input.append(l.split(" "))
    print(input)
    print("Part 1: " + str(part1(input)))
    print("Part 2: " + str(part2(input)))

def part1(input):
    sum = 0
    for i in input:
        if i[1] == 'X':
            sum += 1
            if i[0] == 'A':
                sum += 3
            elif i[0] == 'C':
                sum += 6
        
        elif i[1] == 'Y':
            sum += 2
            if i[0] == 'A':
                sum += 6
            elif i[0] == 'B':
                sum += 3
        
        elif i[1] == 'Z':
            sum += 3
            if i[0] == 'B':
                sum += 6
            elif i[0] == 'C':
                sum += 3
    return sum

def part2(input):
    sum = 0
    for i in input:
        if i[1] == 'X':
            if i[0] == 'A':
                sum += 3 + 0
            elif i[0] == 'B':
                sum += 1 + 0
            elif i[0] == 'C':
                sum += 2 + 0
        
        elif i[1] == 'Y':
            if i[0] == 'A':
                sum += 1 + 3
            elif i[0] == 'B':
                sum += 2 + 3
            elif i[0] == 'C':
                sum += 3 + 3
        
        elif i[1] == 'Z':
            if i[0] == 'A':
                sum += 2 + 6
            elif i[0] == 'B':
                sum += 3 + 6
            elif i[0] == 'C':
                sum += 1 + 6
    return sum

if __name__ == "__main__":
    sys.exit(main())